concourse/pipelines/create-bosh-concourse.yml

```diff
@@ -974,63 +974,115 @@ jobs:
-            SYSTEM_DNS_ZONE_NAME: ((system_dns_zone_name))
-            BOSH_FQDN: ((bosh_fqdn))
-            BOSH_FQDN_EXTERNAL: ((bosh_fqdn_external))
-            BOSH_INSTANCE_PROFILE: ((bosh_instance_profile))
+            - name: bosh-vars-store-cleaned
           run:
             path: sh
             args:
-              - -e
-              - -c
-              - |
-                mkdir -p certs
-                tar -xvzf bosh-ca/bosh-CA.tar.gz -C certs
+            - -c
+            - -e
+            - |
+              cp bosh-vars-store/bosh-vars-store.yml bosh-vars-store-cleaned/bosh-vars-store.yml
 
-                paas-bootstrap/manifests/bosh-manifest/scripts/generate-manifest.sh \
-                  > bosh-manifest-pre-vars/bosh-manifest-pre-vars.yml
+              paas-bootstrap/manifests/bosh-manifest/scripts/cleanup-bosh-vars-store.rb \
+                bosh-vars-store-cleaned/bosh-vars-store.yml
 
-                cp bosh-vars-store/bosh-vars-store.yml bosh-vars-store-updated/bosh-vars-store.yml
+      - task: render-bosh-manifest-pre-vars
+        # i.e. a copy of the manifest before variable interpolation has
+        # taken place, with a `variables:` section still present
+        file: paas-bootstrap/concourse/tasks/render-bosh-manifest.yml
+        params:
+          USE_VARS_STORE: false
+          DEPLOY_ENV: ((deploy_env))
+          AWS_ACCOUNT: ((aws_account))
+          AWS_DEFAULT_REGION: ((aws_region))
+          SYSTEM_DNS_ZONE_NAME: ((system_dns_zone_name))
+          BOSH_FQDN: ((bosh_fqdn))
+          BOSH_FQDN_EXTERNAL: ((bosh_fqdn_external))
+          BOSH_INSTANCE_PROFILE: ((bosh_instance_profile))
+        output_mapping:
+          bosh-manifest: bosh-manifest-pre-vars
 
-                paas-bootstrap/manifests/bosh-manifest/scripts/cleanup-bosh-vars-store.rb \
-                  bosh-vars-store-updated/bosh-vars-store.yml
+      - try:
+          task: check-bosh-leaf-certs
+          config:
+            platform: linux
+            image_resource: *ruby-slim-image-resource
+            inputs:
+              - name: paas-bootstrap
+              - name: bosh-manifest-pre-vars
+              - name: bosh-vars-store-cleaned
+            outputs:
+              - name: bosh-vars-store-rotated
+            run:
+              path: sh
+              args:
+                - -e
+                - -c
+                - |
+                  # this isn't particularly sophisticated - *any* leaf certificate
+                  # being due for rotation will cause them *all* to be regenerated
+                  ./paas-bootstrap/manifests/shared/scripts/check-certificates.rb \
+                    --skip-ca-certs \
+                    bosh-vars-store-cleaned/bosh-vars-store.yml
+
+                  # a no-op "rotation" for the successful case
+                  cp bosh-vars-store-cleaned/bosh-vars-store.yml bosh-vars-store-rotated/bosh-vars-store.yml
+          on_failure:
```

Comment:
Can you explain this setup? It looks to me like it's expecting a non-zero exit code before it rotates the certificates?

Reply:
Yeah I should add a comment. `check-certificates.rb` returns nonzero if it finds any expired certs.

Reply:
better?

Reply:
I think I'm still a bit confused why we're setting it up like a try .. catch. Why not

```
if ! ./paas-bootstrap/manifests/shared/scripts/check-certificates.rb --skip-ca-certs bosh-vars-store-cleaned/bosh-vars-store.yml; then
    # Rotate vars
fi
```

Or something to that effect. If I saw that job failing in concourse, I wouldn't assume it was supposed to. I would assume it was broken. I'm not saying we definitely should change it, just that I don't yet understand the rationale for it as it is.

Reply:
:thinking: that's a good point. I can't remember why I did it like this... it might have just been that I was intending to make the `check-certificates` task a reusable task shared with the standalone `check-certificates` job. But clearly that won't work because it uses different options. So yeah I'll look into it when I get time...

Reply:
Better?

Reply:
Yes, much better I think